Apps/WebClient/src/ClientApp/src/services/winstonLogger.ts

```diff
@@ -6,7 +6,27 @@ import { ExternalConfiguration } from "@/models/configData";
 @injectable()
 export class WinstonLogger implements ILogger {
     private logger!: Logger;
-
+    public addLogging: any = (callerName: string, fn: any) => (
+        ...args: any[]
+    ) => {
+        this.logger.debug(
+            `entering ${callerName}.${fn.name}: input args = ${args}`
+        );
+        try {
+            const valueToReturn = fn(...args);
+            this.logger.debug(
+                `exiting ${callerName}.${fn.name}: valueToReturn = ${
```

Comment:
this looks too close to tracing, I think we don't need it right now

Reply:
right, which is why it is logged at the debug level. Btw, the exception handling block is interesting. On the side note, turning on this code helps to understand the performance of our code as I'm seeing the getHeadingDate method is being call over 10 time. Without this verbose/debug logging code, the performance issue like this one is hidden.